README.md
# Aragon Open Data 3 - Web

Aragon Open data sigue innovando y para su tercera version implementa una version web basada en tecnologias modernas y dinamicas que añaden utilidades al usuario final simplificando el uso de la misma y ofreciendo el contenido de forma mas rapida y efectiva.

##¿Que teconologias usamos?

###Dart
Es la base de **AOD3**, todo esta construido sobre este lenguaje desde el servidor hasta la web ( excluyendo CKAN parte indispensabe en cualquier portal de datos abiertos).

Se ha decidido usar esta tecnologia ya que nos permite usar el mismo lenguaje en la parte servidora como en la cliente pudiendo generar un unico stack de aplicaciones conjuntas que pueden compartir codigo entre ellas sin ningun tipo de problema, ademas de ser apollado por google siendo usado en aplicaciones que generan millones en ingresos proporcianandonos la seguridad de que el lenguaje esta cuidado, es rapido y eficaz en sus funciones.

Tambien es un lenguaje Open Source donde podemos contribuir a sus desarrollo continuando asi nuestro apollo al Open Source desde Aragon Open Data
 
Puedes leer mas sobre Dart aqui: [DART][1]
 
###Angular 2
Como framework para la parte web decidimos usar Angular 2, un framework creado y usado por Google en proyectos criticos, que nos permite crear una pagina SPA (Single Page Aplication) con la cual servir todos nuestros servicios de una forma dinamica y mucho mas rapida que el resto de tecnologias ya que solo tendremos que cargar todo una unica solo vez.

Puedes leer mas sobre Angular 2 aqui: [ANGULAR 2][2]

###Bourbon - Sass
Para los estilos hemos utilizado un framework llamada Bourbon que usa Sass una tecnologia derivada de css que nos permite usar mas herramientas para una mejor edicion y mantenimiento de codigo mientras seguimos sirviendo css para que cualquier navegador pueda comprenderlo

Puedes leer mas sobre Bourbon aqui: [BOURBON][3]

Puedes leer mas sobre Sass aqui: [SASS][4]

##Instalación y primera ejecución

1. Debemos instalar Dart en nuestro sistema operativo, puedes encontrar las guias espedificas para tu SO desde este [LINK][5] 
2. Descargamos el repositorio
```
git clone https://github.com/aragonopendata/Aragon-Open-Data-3.git
```
3. Obtenemos las dependencias necesarias
```
pub get
```
3.1 Instalamos Saas como se describe en esta URL
http://crocodillon.com/blog/how-to-install-sassc-and-libsass-on-ubuntu
OJO: Mirar en el repositorio de cada librería la versión que corresponde en el branch


4. Ejecutamos la aplicaciones
```
pub serve --port=8080 --mode=debug
```
*Si aparece algún tipo de error al acceder a localhost:8001, probar a realizar los siguientes pasos:
i) apt-get remove dart
ii) apt-get install dart=1.21.*

Si tenemos cualquier tipo de test también podemos ejecutar
```
pub run test
```

Ahora tendremos la aplicacion en local y podremos visitarla desde el puerto 8080 en local

##Sacar la version a Producción

1.Ejecutamos los test para comprobar que todo fnciona correctamente ( si tenemos tests )
```
pub run test
```
2.Buildeamos la aplicacion
```
pub build --mode=release
```

Este comando nos generara un montón de archivos en una carpeta llamada build, simplemente movemos esa carpeta a la raíz de nuestro servidor  y podremos acceder a ella.

Es recomendable añadir esta configuración a apache para que nos carguen los links de las rutas.
```
RewriteEngine On  
  # If an existing asset or directory is requested go to it as it is
  RewriteCond %{DOCUMENT_ROOT}%{REQUEST_URI} -f [OR]
  RewriteCond %{DOCUMENT_ROOT}%{REQUEST_URI} -d
  RewriteRule ^ - [L]

  # If the requested resource doesn't exist, use index.html
  RewriteRule ^ /index.html
``` 
 
 [1]: https://www.dartlang.org/
 [2]: https://webdev.dartlang.org/angular
 [3]: http://bourbon.io/
 [4]: http://sass-lang.com/
 [5]: https://www.dartlang.org/install

CHANGELOG.md
# Change Log
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/)
and this project adheres to [Semantic Versioning](http://semver.org/).

## [Unreleased]

## [3.0.6] - 2017-08-31
### Changed
- Aragopedia 2.0 API iframe fixed.

## [3.0.5] - 2017-07-25
### Changed
- Flat rent application added.

## [3.0.4] - 2017-06-01
### Changed
- New 'Términos y condiciones' page added.

## [3.0.3] - 2017-05-29
### Added
- New 'Términos y condiciones' page added.

### Changed
- Fixed links to 'Social Data' page.
- Fixed link to 'Términos y condiciones' page.
- Several text corrections.

## [3.0.2] - 2017-04-12
### Added
- Added space chars to text with a bad format in "Info" section.

## [3.0.1] - 2017-04-07
### Added
- Search box section at the top of page.
- New CHANGELOG file necessary to follow the changes of the application in the new version system.

## [3.0.0] - 2017-03-30
### Added
- Initial version of the new version system of the application.
- Home based on Angular2 + Dart.

[Unreleased]: https://github.com/aragonopendata/Aragon-Open-Data-3/compare/master...develop
[3.0.6]: https://github.com/aragonopendata/Aragon-Open-Data-3/compare/v3.0.5...v3.0.6
[3.0.5]: https://github.com/aragonopendata/Aragon-Open-Data-3/compare/v3.0.4...v3.0.5
[3.0.4]: https://github.com/aragonopendata/Aragon-Open-Data-3/compare/v3.0.3...v3.0.4
[3.0.3]: https://github.com/aragonopendata/Aragon-Open-Data-3/compare/v3.0.2...v3.0.3
[3.0.2]: https://github.com/aragonopendata/Aragon-Open-Data-3/compare/v3.0.1...v3.0.2
[3.0.1]: https://github.com/aragonopendata/Aragon-Open-Data-3/compare/v3.0.0...v3.0.1
[3.0.0]: https://github.com/aragonopendata/Aragon-Open-Data-3/releases/tag/v3.0.0


lib/service/home_service.dart
import 'package:angular2/core.dart';
import 'package:aod3/object/home_element.dart';
import 'package:aod3/object/home_card.dart';

///Servicio que se encarga de abastecer a [HomeComponent] de todos los datos necesarios para poder mostrar las tarjetas
@Injectable()
class HomeService {
  ///Devuelve un listado de todas las tarjetas agrupadas en sus determinados contenedores
  ///
  ///1.Datos
  ///2.Servicios
  ///3.Informacion
  ///4.Herramientas
  List<HomeElement> getList() {
    return <HomeElement>[
      new HomeElement(
          'Search',
          '',
          <HomeCard>[])..customHtml = "dataSearch",
      new HomeElement(
          'Datos',
          'Lo más importante de Aragón Open Data son sus datos y los hemos ordenado para que puedas encontrarlos fácilmente.',
          <HomeCard>[
            new HomeCard(
                'Banner-01-BancoDatos',
                'Banco de datos',
                'Además de buscar, puedes acceder a los datos mediante categorías y filtros.',
                '/datos/catalogo.html',
                false),
            new HomeCard(
                'Banner-02-Temas',
                'Temas',
                'Puedes llegar a los datos mediante la navegación temática.',
                '/datos/tema',
                false),
            new HomeCard(
                'Banner-03-Publicadores',
                'Publicadores',
                'Conoce qué organizaciones aportan sus datos a Aragón Open Data.',
                '/datos/organizacion',
                false),
          ]),
      new HomeElement(
          'Servicios',
          'En Aragón Open Data hemos creado servicios para ofrecer los datos, reutilizarlos y acceder de forma sencilla a la información.',
          <HomeCard>[
            new HomeCard(
                'Banner-05-Aragopedia',
                'Aragopedia',
                'Entiende y usa los datos abiertos de Aragón Open Data a través de los municipios y comarcas de Aragón.',
                '/servicios/aragopedia',
                false),
            new HomeCard(
                'Banner-06-Presupuestos',
                'Presupuestos',
                'Conoce de un modo claro cómo se distribuye el presupuesto de Aragón y de dónde vienen los ingresos y a qué destinamos el gasto.',
                'http://presupuesto.aragon.es/',
                false),
            new HomeCard(
                'Banner-04-OpenSocialData',
                'Open Social Data',
                'Escucha, visualiza y analiza las conversaciones sobre Aragón en las redes sociales y conoce qué está pasando.',
                '/servicios/open-social-data',
                false),
            new HomeCard(
                'Banner-07-CRAs',
                'CRAs Aragón',
                'Descubre más sobre los Colegios Rurales Agrupados de Aragón.',
                '/servicios/cras',
                false),
          ]),
      new HomeElement(
          'Informacion',
          'Te explicamos qué es Aragón Open Data, qué hacemos y cómo puedes sumarte y colaborar.',
          <HomeCard>[
            new HomeCard(
                'Banner-10-Informacion',
                'Información',
                'Conoce más sobre el entorno open data y qué hacemos en Aragón.',
                '/info/open-data',
                true),
            new HomeCard(
                'Banner-09-Aplicaciones',
                'Aplicaciones',
                'Otras aplicaciones que aprovechan y reutilizan nuestros datos para ofrecértelos.',
                '/info/aplicaciones',
                true),
            new HomeCard(
                'Banner-11-Eventos',
                'Eventos',
                'Agenda y actividades de Aragón Open Data.',
                '/info/eventos',
                true),
            new HomeCard(
                'Banner-08-Colabora',
                'Colabora',
                'Sugiere datos, formatos y utilidades que te gustaría disponer de Aragón Open Data.',
                '/info/colabora',
                true),
          ]),
      new HomeElement(
          'Herramientas',
          'Aprende y consigue nuestros contenidos, datos y código con estas utilidades hechas para ti.',
          <HomeCard>[
            new HomeCard(
                'Banner-12-Campus',
                'AOD Campus',
                'Contenido ofrecido para que aprendas y te formes a través de nuestros materiales y tutoriales.',
                '/campus',
                true),
            new HomeCard(
                'Banner-13-Desarrolladores',
                'Desarrolladores',
                'Estándares, interoperabilidad, estructuras y herramientas en las que se sustenta Aragón Open Data.',
                '/herramientas/desarrolladores',
                true),
            new HomeCard(
                'Banner-14-APIs',
                'APIs',
                'Tienes APIs para la búsqueda de datos, para AragoPedia y para el servicio Open Social Data.',
                '/herramientas/apis',
                true),
            new HomeCard(
                'Banner-15-SPARQL',
                'SPARQL',
                'Obtén datos e información de la Aragopedia a través de este punto de consulta.',
                '/herramientas/sparql',
                true),
            new HomeCard(
                'Banner-16-GITHUB',
                'GITHUB',
                'Código y fuentes usadas para desarrollar Aragón Open Data lo tienes disponible en nuestro perfil de GitHub.',
                'https://github.com/aragonopendata',
                false),
          ]),
    ];
  }
}
